iceoryx_hoofs/test/moduletests/test_cxx_helplets.cpp

```diff
@@ -507,13 +513,163 @@ TEST(Helplets_test_isValidFilePath, WhenOneInvalidCharacterIsContainedPathIsInva
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/../some/dir/gimme-blubb")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/../some/dir/gimme-blubb/./dir/")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/./blubb/dir/gimme-blubb")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/./blubb/dir/gimme-blubb/../dir/")));
+}
+
+TEST(Helplets_test_isValidPath, SingleEntryIsValidPath)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "6983ab77-d658-408d-97aa-bd1d218560fb");
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("gimme-blubb")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("gimme-blubb/")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("a")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("a/")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("fuu:blubb")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("fuu:blubb/")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/blarbi")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/blarbi/")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/x")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/x/")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/fuu:-012")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/fuu:-012/")));
+}
+
+TEST(Helplets_test_isValidPath, ValidPathsWithNoRelativeComponentAreValid)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "bf7a0a75-c59e-46a8-96f1-1f848e1c3e43");
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/fuu/bla/blubb/balaa")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/fuu/bla/blubb/")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/a/b/c/d/1/2/4")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/a/b/c/d/1/2/")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("asd/fuu/asdaaas/1")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("asd/fuu/asdaaas/")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("123/456")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("123/456/")));
+}
+
+TEST(Helplets_test_isValidPath, EndingWithRelativePathComponentIsInvalid)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "506f9823-39cc-4cbc-b064-84d45b2311e8");
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("/..")));
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("/.")));
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("./..")));
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("../.")));
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("some/path/to/..")));
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("/another/path/to/.")));
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("../bla/fuu/../blubb/.")));
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("./blubb/fuu/../bla/..")));
+}
+
+TEST(Helplets_test_isValidPath, FilePathsWithEndingDotsAreInvalid)
```

Comment:
:grimacing:

Reply:
So I splitted the `isValidFilePath` function up so that I can reuse parts of it for `isValidPath`. The new function is called now `isValidPathEntry`. Since the function names started to confuse me I renamed them slightly:
* `isValidFilePath` -> `isValidPathToFile`
* `isValidPath` -> `isValidPathToDirectory`